Sources/AppStoreConnectCLI/Services/SyncResourceComparator.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import Foundation
+
+enum SyncStrategy<T: SyncResultRenderable> {
+    case delete(T)
+    case create(T)
+    case update(T)
+}
+
+protocol SyncResourceProcessable: SyncResourceComparable, SyncResultRenderable { }
+
+protocol SyncResourceComparable: Hashable {
+    associatedtype T: Comparable
+
+    var compareIdentity: T { get }
+}
+
+struct SyncResourceComparator<T: SyncResourceProcessable> {
+
+    let localResources: [T]
+    let serverResources: [T]
+
+    private var localResourcesSet: Set<T> { Set(localResources) }
+    private var serverResourcesSet: Set<T> { Set(serverResources) }
+
+    func compare() -> [SyncStrategy<T>] {
```

Comment:
🤔 I wonder if it would make use to use `CollectionDifference`.

```
var x = [2,3,4]
var y = [1,2,3]

let d = x.difference(from: y)
d.insertions
d.removals
```

Though that has no concept of a change. Ie, the properties of a value being different but the identity being the same.

Reply:
@orj CollectionDifference can tell the difference when two collections are **ordered**, but for our cases, we can't guarantee that the resources read from disk and server are well-sorted or ordered. So in here, I used Set() what do you think ?